### PR TITLE
Adding ability to work with asynchronous tests

### PR DESCRIPTION
**Pull request by therealparmesh**
See [Mocha's documentation on testing asynchronous code](https://mochajs.org/#asynchronous-code) for the detailed description behind what I implemented here. Please let me know your thoughts on this!

### CODE REVIEW

**Comment by mthuret** [comment]
That's seems super cool. Thanks so much!

Are you using some formatter like prettier? Because the diff is a bit hard to follow :) 

Also could you fix the conflict before I merge this :)

**Comment by therealparmesh** [comment]
I'll clean up the diff and fix the conflict!

**Comment by therealparmesh** [comment]
We're good to go! I rebased onto your updated `master` branch.

FYI: The invalid peer dependency issue that I attempted to fix was already fixed in a separate pull request. This was why this pull request had merge conflicts.
